ql/src/java/org/apache/hadoop/hive/ql/parse/TaskCompiler.java

```diff
@@ -134,6 +135,16 @@ public void compile(final ParseContext pCtx,
     boolean isCStats = pCtx.getQueryProperties().isAnalyzeRewrite();
     int outerQueryLimit = pCtx.getQueryProperties().getOuterQueryLimit();
 
+    boolean directInsertCtas = false;
+    if (pCtx.getCreateTable() != null && pCtx.getCreateTable().getStorageHandler() != null) {
+      try {
+        directInsertCtas =
+            HiveUtils.getStorageHandler(conf, pCtx.getCreateTable().getStorageHandler()).directInsertCTAS();
+      } catch (HiveException e) {
```

Comment:
When do we get `HiveException`?

Reply:
We get it in case of a `ClassNotFoundException` is thrown inside `HiveUtils.getStorageHandler`. That would happen if the storage handler class is not available on the classpath